scripts/lib/libvirt/libvirt_setup.py

```diff
@@ -106,6 +106,25 @@ def compute_config(args, cpu_flags=cpuflags()):
     else:
         nodememory = args.controllernodememory
 
+    raidvolume = ""
+    if args.raidvolumenumber and args.raidvolumenumber > 1:
+        for i in range(1, int(args.raidvolumenumber)):
+            raid_template = string.Template(readfile(
+                "{0}/extra-volume.xml".format(TEMPLATE_DIR)))
+            raid_values = dict(
+                volume_serial="{0}-node{1}-raid{2}".format(
+                    args.cloud,
+                    args.nodecounter,
+                    i),
+                source_dev="{0}/{1}.node{2}-raid{3}".format(
+                    args.vdiskdir,
+                    args.cloud,
+                    args.nodecounter,
+                    i),
+                target_dev=targetdevprefix + ''.join(alldevices.next()),
```

Comment:
This is when a test comes handy.  Not sure what do you expect, but the second time that you create this dictionary, the iterator will be empty.  So I guess that there is a bug there and in the source.

Can create some test?